### DISCUSSION TITLE
Error "NumericValueOutOfRange integer out of range" in scheduler, task_insance.priority_weight column overflow

### DISCUSSION THREAD

**Discussion by kosteev** [apache/airflow#22957] [General]
### Apache Airflow version

2.2.3

### What happened

Scheduler fails with following exception in case of priority_weight_total method returning out of integer range value.

"Original exception was: (psycopg2.errors.NumericValueOutOfRange) integer out of range"

### How to reproduce

DAG to reproduce issue.

<pre>
import airflow
from airflow import DAG
from airflow.operators.bash_operator import BashOperator
from datetime import timedelta

default_args = {
    'start_date': airflow.utils.dates.days_ago(0),
    'retries': 1,
    'retry_delay': timedelta(minutes=5)
}

dag = DAG(
    'monitoring_sched',
    default_args=default_args,
    description='liveness monitoring dag',
    schedule_interval='*/20 * * * *',
    dagrun_timeout=timedelta(minutes=120))

# priority_weight has type int in Airflow DB, uses the maximum.

# priority_weight has type int in Airflow DB, uses the maximum.
t1 = BashOperator(
    task_id='echo',
    bash_command='echo test',
    dag=dag,
    queue='kubernetes',
    depends_on_past=False,
    priority_weight=2**31-1)

# priority_weight has type int in Airflow DB, uses the maximum.
t2 = BashOperator(
    task_id='sleep',
    bash_command='sleep 1900',
    queue='kubernetes',
    dag=dag,
    depends_on_past=False,
    priority_weight=2**31-1)

# priority_weight has type int in Airflow DB, uses the maximum.
t3 = BashOperator(
    task_id='ech2o',
    bash_command='echo done',
    dag=dag,
    queue='kubernetes',
    depends_on_past=False,
    priority_weight=2**31-1)

t1 >> t2 >> t3
</pre>

### Operating System

Composer

### Deployment

Composer

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
Surely you should not do that. There are about a million ways you can make application that parses python fail by providing bad code.

**Reply by kosteev** [reply]
We had a discussion in https://github.com/apache/airflow/pull/22784 and got to the point that we want to have check on DAG import.

The issue here is that scheduler starts to fail and become unhealthy and in the meantime DAG definition code is correct (priority_weight parameter for each task doesn't go beyond integer boundaries, it is that priority_weight_total goes and this is what used as value for task_instance record).

**Reply by potiuk** [reply]
I see. Surely we can protect against that  case indeed.
